City-of-Helsinki/berth-reservations-admin
# VEN-985 | Fix select rows in paginated tables

## Description :sparkles:
- Implemented a custom hook `usePreserveSelect` for storing table selection in Recoil
- Fix non-updated state bug in `ListActions` component and remove redundant args
- Add usePreserveSelect to ApplicationList
- Add usePreserveSelect to UnmarkedWsNoticeList

## Issues :bug:

### Closes :no_good_woman:
[**VEN-985**](https://helsinkisolutionoffice.atlassian.net/browse/VEN-985): FE - Fix multi-select rows on paginated tables

### Related :handshake:

## Testing :alembic:

### Automated tests :gear:️
https://github.com/City-of-Helsinki/berth-reservations-admin/blob/1de9c1814c67612ef7649d4dd840e5682dc39282/src%2Fcommon%2Futils%2F__tests__%2FusePreserveSelect.test.tsx#L21

### Manual testing :construction_worker_man:
- Go to `Venepaikkahakemukset` view
- Select some rows from different pages in the table
- To prove that the selection is preserved you can select an action like `Lähetä tarjous` and it should perform the action on all the selected rows
- Going back and forth between pages should NOT affect the selection in `Venepaikkahakemukset` page
- The same behavior is also implemented in the `Nostojärjestysilmoitukset` view.

## Screenshots :camera_flash:

## Additional notes :spiral_notepad:
